apps/wallet-ui/src/components/NftCard.tsx

```diff
@@ -0,0 +1,60 @@
+"use client";
+import type { Erc721Token } from "@/types/Erc721Token";
+import { useQuery } from "@tanstack/react-query";
+import Image from "next/image";
+import React from "react";
+import { shortenAddress } from "thirdweb/utils";
+import { getChain } from "../lib/chains";
+import { ChainIcon } from "./ChainIcon";
+import { NftModal } from "./NftModal";
+import { Badge } from "./ui/badge";
+import Card from "./ui/card";
+import { Skeleton } from "./ui/skeleton";
+
+export function NftLoadingCard() {
+  return <Skeleton className="w-[300px] h-[300px]" />;
+}
+
+export default function NftCard({
+  data,
+}: { data: Erc721Token; width?: number; height?: number }) {
+  const { data: chainData } = useQuery({
+    queryKey: ["chain-metadata", data.chainId],
+    queryFn: async () => {
+      return getChain(data.chainId.toString());
+    },
+    enabled: !!data.chainId,
+  });
```

Comment:
what chain data do you need, if it's icon etc there are specific hooks that efficiently fetch it in `thirdweb/react` I believe (used in connect ui) - should be able to use those (might have to export them first)

Reply:
or if this could be SSR then fetch the chain metadata during SSR